pkg/guard-gate/state.go

```diff
@@ -131,14 +132,24 @@ func (gs *gateState) profileAndDecidePod() {
 		if decision != nil {
 			gs.addStat("PodAlert")
 			gs.alert = decision.String("Pod  -> ")
-
-			logAlert(gs.alert)
-			// terminate the reverse proxy
-			gs.cancelFunc()
+			gs.logAlert()
+			if gs.shouldBlock() {
```

Comment:
Cancel the queue-proxy only if we are in Block mode (and the pod shows irregular behavior)